hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -370,9 +388,15 @@ private IOEngine getIOEngineFromName(String ioEngineName, long capacity)
       // In order to make the usage simple, we only need the prefix 'files:' in
       // document whether one or multiple file(s), but also support 'file:' for
       // the compatibility
-      String[] filePaths =
+      filePaths =
           ioEngineName.substring(ioEngineName.indexOf(":") + 1).split(FileIOEngine.FILE_DELIMITER);
-      return new FileIOEngine(capacity, filePaths);
+      hasKey = FileIOEngineUtils.readLongFormPersistencePath(persistencePath) == -1;
+      if (hasKey) {
+        return new FileIOEngine(algorithm, persistencePath, capacity, filePaths);
+      } else {
+        return new FileIOEngine(capacity,filePaths);
```

Comment:
I don't like the idea of keeping the older constructor, the `hasKey` check can put into the new constructor as well.